From 8a7db2e4ed7f986388c4b69bb130414defbc2c6b Mon Sep 17 00:00:00 2001
From: Quentin Casasnovas <quentinx.casasnovas@intel.com>
Date: Thu, 14 Mar 2013 10:47:59 +0100
Subject: [PORT FROM MAIN]wifi: check fwReloadSoftap to properly enable SoftAp

BZ: 92804

fwReloadSoftap was blindly telling the HAL to switch to SoftAp when this
function can be used to go back from Softap to STA.

Change-Id: Ia09465e38a58c2609de5efdf03f5dfcac5ebbc10
Orig-Change-Id: Id9a1e3bcd535f401664ab4fc26df170e7fc4da6e
Category: device enablement
Domain: CWS.WIFI-Common
Origin: upstream
Upstream-Candidate: yes
Signed-off-by: Quentin Casasnovas <quentinx.casasnovas@intel.com>
---
 SoftapController.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/SoftapController.cpp b/SoftapController.cpp
index 47904d6..995e1de 100644
--- a/SoftapController.cpp
+++ b/SoftapController.cpp
@@ -253,7 +253,12 @@ int SoftapController::fwReloadSoftap(int argc, char *argv[])
     iface = argv[2];
     stopDriver(iface);
 
-    wifi_switch_driver_mode(WIFI_AP_MODE);
+    if (!strcmp("STA", argv[3]))
+        wifi_switch_driver_mode(WIFI_STA_MODE); /* which is STA + P2P... */
+    else if (!strcmp("AP", argv[3]))
+        wifi_switch_driver_mode(WIFI_AP_MODE);
+    else if (!strcmp("P2P", argv[3]))
+        wifi_switch_driver_mode(WIFI_P2P_MODE);
 
     /**
      * Sleep to workaround issue in the brcm driver which is tracked by BZ
-- 
1.7.4.1

